nullaway/src/test/java/com/uber/nullaway/NullAwayJSpecifyGenericsTests.java

```diff
@@ -198,6 +198,175 @@ public void downcastInstantiation() {
         .doTest();
   }
 
+  @Test
+  public void genericsChecksForAssignments() {
+    makeHelper()
+        .addSourceLines(
+            "Test.java",
+            "package com.uber;",
+            "import org.jspecify.annotations.Nullable;",
+            "class Test {",
+            "    static class NullableTypeParam<E extends @Nullable Object> {}",
+            "    static void testBadNonNull(NullableTypeParam<@Nullable String> t1) {",
+            "   // BUG: Diagnostic contains: Cannot assign from type",
+            "        NullableTypeParam<String> t2 = t1;",
```

Comment:
Re: the formatting, the Error Prone Support project has a checker for this:

https://error-prone.picnic.tech/bugpatterns/ErrorProneTestHelperSourceFormat/#ErrorProneTestHelperSourceFormat

Not for this PR but it might be worth enabling in the future.

Reply:
Follow-up: unfortunately the auto-fixes from that other check end up doing things some things we don't want and actually cause some tests to fail.  All potentially fixable, but not worth it right now.

Reply:
@msridhar thanks for trying it! The check is rather opinionated, but if there are possible tweaks we can make to it, do let us know :)

Reply:
@Stephan202 I opened https://github.com/PicnicSupermarket/error-prone-support/issues/474 🙂 Thanks for the help!

Reply:
@NikitaAware unfortunately a lot of the formatting here is still off (we prefer 2-space indentation like Google Java Format does).  Let me try to fix the formatting for you later today; will push a commit